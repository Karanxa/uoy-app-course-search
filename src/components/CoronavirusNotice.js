const CoronavirusNotice = () => {
    return (
        <div id="global-notice">
            <h2>
                Stay <em>smart</em>, <em>supportive</em> and <em>safe</em>
            </h2>
            <p>
                <a href="https://coronavirus.york.ac.uk/">Coronavirus information for staff and students</a>
            </p>
        </div>
    );
};

export { CoronavirusNotice };
