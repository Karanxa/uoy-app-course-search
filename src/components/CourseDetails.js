import React from "react";
import { COURSE_MODEL } from "../constants/CourseModel";
import { TagList } from "./TagList";
import { Tag } from "./Tag";
import PropTypes from "prop-types";

const CourseDetails = ({ course }) => {
    return (
        <TagList>
            <LevelAndAwardTag level={course.level} award={course.award} />
            <StartDateAndLengthTag yearOfEntry={course.yearOfEntry} length={course.length} />
        </TagList>
    );
};

CourseDetails.propTypes = {
    course: COURSE_MODEL,
};

const LevelAndAwardTag = ({ level, award }) => {
    if (level || award) {
        return <Tag topIcon="mortar-board" title={level} subText={award} />;
    }

    return null;
};

LevelAndAwardTag.propTypes = {
    level: PropTypes.string,
    award: PropTypes.string,
};

const StartDateAndLengthTag = ({ yearOfEntry, length }) => {
    if (yearOfEntry || length) {
        const startYear = yearOfEntry ? `Start ${yearOfEntry.slice(0, 4)}` : null;
        return <Tag topIcon="clock-o" title={startYear} subText={length} />;
    }

    return null;
};

StartDateAndLengthTag.propTypes = {
    yearOfEntry: PropTypes.string,
    length: PropTypes.string,
};

export { CourseDetails };
