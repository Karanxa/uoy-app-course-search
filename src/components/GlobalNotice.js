import PropTypes from "prop-types";
import React from "react";

const GlobalNotice = ({ children }) => {
    if (children) {
        return <div className="c-global-notice c-global-notice--highlighted">{children}</div>;
    }

    return null;
};

GlobalNotice.propTypes = {
    children: PropTypes.node,
};

export { GlobalNotice };
