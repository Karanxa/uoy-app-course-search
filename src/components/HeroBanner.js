import PropTypes from "prop-types";
import React from "react";

const HeroBanner = ({ children }) => (
    <figure className="c-hero">
        <div className="c-hero__banner">
            <div className="o-wrapper">
                <div className="o-grid">
                    <div className="o-grid__row">
                        <div className="o-grid__box o-grid__box--full">{children}</div>
                    </div>
                </div>
            </div>
        </div>
    </figure>
);

HeroBanner.propTypes = {
    children: PropTypes.node,
};

export { HeroBanner };
