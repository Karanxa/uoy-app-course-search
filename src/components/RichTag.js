import PropTypes from "prop-types";
import React from "react";

const RichTag = ({ topIcon, title, subText }) => (
    <li className="rich-tag">
        {topIcon && <i aria-hidden className={`c-icon c-icon--${topIcon} c-icon--2x`} data-testid="tag-icon" />}
        <p className="rich-tag__title">{title}</p>
        <p className="rich-tag__text">{subText}</p>
    </li>
);

RichTag.propTypes = {
    topIcon: PropTypes.string,
    title: PropTypes.string,
    subText: PropTypes.string,
};

const ArrayTag = ({ title, content }) => {
    const contents = content.map((item) => (
        <ArrayTagContent key={item.text} icon={item.icon} label={item.text} fullWidth={title === null} />
    ));

    return (
        <li className="rich-tag">
            {title && <p className="rich-tag__title">{title}</p>}
            <ul className="c-icon--ul">{contents}</ul>
        </li>
    );
};

ArrayTag.propTypes = {
    title: PropTypes.string,
    content: PropTypes.array,
};

const ArrayTagContent = ({ icon, label }) => (
    <li className="rich-tag__text">
        {icon && <i aria-hidden className={`c-icon c-icon--${icon} c-icon--1g c-icon--li`} data-testid={icon} />}
        {label}
    </li>
);

ArrayTagContent.propTypes = {
    icon: PropTypes.string,
    label: PropTypes.string,
};

export { RichTag, ArrayTag };
