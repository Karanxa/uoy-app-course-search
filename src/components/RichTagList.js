import React from "react";
import PropTypes from "prop-types";

const RichTagList = (props) => {
    return <ul className="rich-tag-list">{props.children}</ul>;
};

RichTagList.propTypes = {
    children: PropTypes.node,
};

export { RichTagList };
