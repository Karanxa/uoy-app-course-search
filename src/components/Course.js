import React from "react";
import { COURSE_MODEL } from "../constants/CourseModel";
import { CourseDetails } from "./CourseDetails";

const Course = ({ course }) => {
    const title = () => {
        if (course.award) {
            return `${course.title} - ${course.award}`;
        }

        return course.title;
    };

    const linkId = "course-title-" + title(course).replace(/ /g, "-");

    return (
        <li className="c-listings-item ">
            <a className="c-listings-item__link" aria-labelledby={linkId} href={course.liveUrl}>
                <h2 id={linkId} className="c-listings-item__title">
                    {title(course)}
                </h2>
                <div className="c-listings-item__description">
                    <CourseDetails course={course} />
                </div>
            </a>
        </li>
    );
};

Course.propTypes = {
    course: COURSE_MODEL,
};

export { Course };
