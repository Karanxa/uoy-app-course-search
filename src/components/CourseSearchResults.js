import { Course } from "./Course";
import PropTypes from "prop-types";
import { COURSE_MODEL } from "../constants/CourseModel";
import React from "react";
import { Alert } from "@university-of-york/esg-lib-pattern-library-react-components";
import { noSearchConducted } from "../utils/searchTerms";

const CourseSearchResults = ({ isSuccessfulSearch, searchResults, searchTerm }) => {
    if (noSearchConducted(searchTerm)) {
        return null;
    }

    if (!isSuccessfulSearch) {
        return <SearchFailedMessage />;
    }

    if (!searchResults?.length) {
        return <NoSearchResultsMessage />;
    }

    return (
        <ul className="c-listings">
            {searchResults?.map((course) => (
                <Course key={course.liveUrl} course={course} />
            ))}
        </ul>
    );
};

CourseSearchResults.propTypes = {
    isSuccessfulSearch: PropTypes.bool,
    searchResults: PropTypes.arrayOf(COURSE_MODEL),
    searchTerm: PropTypes.string,
};

const SearchFailedMessage = () => (
    <Alert className="warning">
        Course search is currently unavailable. Please try again later, or{" "}
        <a href="https://www.york.ac.uk/it-support/">contact IT Support</a>.
    </Alert>
);

const NoSearchResultsMessage = () => <Alert className="info">Sorry, your search did not return any results.</Alert>;

export { CourseSearchResults };
