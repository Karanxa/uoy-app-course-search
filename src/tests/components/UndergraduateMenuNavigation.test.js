import { render } from "@testing-library/react";
import { UndergraduateMenuNavigation } from "../../components/UndergraduateMenuNavigation";

describe("UndergraduateMenuNavigation", () => {
    it("renders as expected", () => {
        const { asFragment } = render(<UndergraduateMenuNavigation />);

        expect(asFragment()).toMatchSnapshot();
    });
});
