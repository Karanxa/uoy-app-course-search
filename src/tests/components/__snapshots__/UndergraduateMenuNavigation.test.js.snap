// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndergraduateMenuNavigation renders as expected 1`] = `
<DocumentFragment>
  <nav
    aria-label="Main Navigation"
    class="c-nav c-nav--main"
    id="Main-Navigation"
    role="navigation"
  >
    <h4
      class="c-nav__header"
    >
      Courses 2021/22
    </h4>
    <ul
      class="c-nav__list c-nav__list c-nav__list--associative"
    >
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/courses/all"
        >
          All courses
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/courses-2021/teaching-learning/"
        >
          Teaching and learning
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/international/applying/pathways/"
        >
          International foundation programme
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/about/term-dates/"
        >
          Term dates
        </a>
      </li>
    </ul>
    <h4
      class="c-nav__header"
    >
      Other sections
    </h4>
    <ul
      class="c-nav__list c-nav__list c-nav__list--structural"
    >
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/courses/"
        >
          Courses
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/applying/"
        >
          Applying
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/visits/"
        >
          Visit us
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/fees-funding/"
        >
          Fees and funding
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/accommodation/undergraduate/"
        >
          Accommodation
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/student-life/"
        >
          Student life
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/careers-skills/"
        >
          Careers and skills
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/study-abroad/"
        >
          Study abroad
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/international/"
        >
          International students
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/prospectus/"
        >
          Prospectus
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/schools-and-colleges/"
        >
          Schools and colleges
        </a>
      </li>
      <li
        class="c-nav__item"
      >
        <a
          class="c-nav__link"
          href="https://www.york.ac.uk/study/undergraduate/why-york/"
        >
          Why York?
        </a>
      </li>
      <li
        class="c-nav__item c-nav__item--more"
      >
        <a
          class="c-nav__link"
        >
          More...
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
